corehq/apps/api/resources/v0_4.py

```diff
@@ -457,7 +459,8 @@ def dehydrate_module(app, module, langs):
     def dehydrate_modules(self, bundle):
         app = bundle.obj
 
-        if app.doc_type == Application._doc_type:
+        # support returning linked applications upon receiving an application list request
+        if app.doc_type == Application._doc_type or app.doc_type == LinkedApplication._doc_type:
```

Comment:
nit: you could have done 

```suggestion
        if app.doc_type in [Application._doc_type, LinkedApplication._doc_type]:
```

Reply:
Made that change. Thanks.